### PR TITLE
Fix errors in example apps

### PR DESCRIPTION
**Pull request by sodic**
Fixes [a few errors](https://github.com/wasp-lang/wasp/pull/720#issuecomment-1246828596) in two of our example apps.

### CODE REVIEW

File: examples/realworld/ext/auth/SignupPage.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ import { errorMessage } from '../utils.js'
...
 const SignupPage = () => {
   const history = useHistory()
-  const [username, setUsername] = useState()
-  const [email, setEmail] = useState()
-  const [password, setPassword] = useState()
-  const [submitError, setSubmitError] = useState()
+  const [username, setUsername] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [submitError, setSubmitError] = useState(null)
```
Comment (sodic) [inline examples/realworld/ext/auth/SignupPage.js:15]: `useState` returns `undefined`. Passing `undefined` as a `value` prop to an input tells it to be uncontrolled. Changing it to a defined value later tells it to be controlled. React doesn't like this. We can fix it by initializing controlled inputs to sensible default values.

File: examples/thoughts/ext/MainPage.js [hunk 2]
```diff
@@ -69,7 +62,6 @@ const Thought = (props) => {
       try {
         await createThought({ textMarkdown: textMd.trim(), tagNames })
         history.push('/thoughts') // TODO: Would be cool if this was type checked somehow or if string was coming from the Wasp API.
-        resetForm()
```
Comment (sodic) [inline examples/thoughts/ext/MainPage.js:72]: Swapping the order between `history.push` and `resetForm` would have also fixed the issue (as would adding a check whether the component was mounted before resetting the state). However, there's no need to reset the state before unmounting, so I removed the entire thing.

File: examples/thoughts/ext/ThoughtsPage.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ const ThoughtListView = (props) => (
   <div className="thought-list-view">
     <div className="thought-list-view-tags">
       {props.thought.tags?.map(tag => (
-        <Tag name={tag.name} />
+        <Tag key={tag.id} name={tag.name} />
```
Comment (sodic) [inline examples/thoughts/ext/ThoughtsPage.js:46]: Found an extra warning :)

File: examples/thoughts/ext/MainPage.js (hunk 2 above)
Comment (shayneczyzewski) [reply examples/thoughts/ext/MainPage.js:72]: Nice!

**Review by shayneczyzewski — APPROVED** [review]
Looks great to me @sodic thanks for fixing these! I think keeping our examples tidy and clean will speak well about Wasp in general, so thanks for taking this on. 🚀

**Pull request merged**